cmake/CMakeLists.txt

```diff
@@ -23,11 +23,14 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CXX_FLAGS)
   set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." FORCE)
 endif(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CXX_FLAGS)
 
-# remove any style headers in the src dir
-file(GLOB SRC_STYLE_FILES ${LAMMPS_SOURCE_DIR}/style_*.h)
-if(SRC_STYLE_FILES)
-  file(REMOVE ${SRC_STYLE_FILES})
-endif()
+# check for files installed by make-based buildsystem 
+file(GLOB SRC_STYLE_FILES ${LAMMPS_SOURCE_DIR}/*/*.cpp)
+foreach(_SRC SRC_STYLE_FILES)
+  get_filename_component(FILENAME "${_SRC}" NAME)
+  if(EXISTS ${LAMMPS_SOURCE_DIR}/${FILENAME})
+    message(FATAL_ERROR "Found packages installed by the make-based buildsystem, please run 'make -C ${LAMMPS_SOURCE_DIR} no-all purge'")
+  endif()
+endforeach()
```

Comment:
@sjplimp @akohlmey as expected that check is really slow!

Why isn't if enough to check for `style_*.h` in `src/`?

Reply:
Because style_*.h files are only created when calling `make`. For instance, they are not present in LAMMPS tarballs, which have some packages installed. Or if you download a snapshot from github or create local repo with git clone and then do make yes-molecule, you will not have the style_*.h files.

Reply:
We could cache the list and result of check and run it only once!

Reply:
Nope. we talked about this yesterday. Somebody may get an 'invalid xxx style' error. search the mailing list, find the recommendation to do 'make yes-pkg' and compile again. There ain't no escape from the blues. :-(

Reply:
True, so back to idea of having a hidden file for each installed packages?

Reply:
And how about caching the list of cpp files in src/ and re-running that changes?

Reply:
Rather than globing all files in all package folders, i would loop over all packages, and loop first over "commonly used" packages first (until we find a trick to enable/disable groups of packages as discussed), we will have a default set of packages installed. we should have 5 groups:
- popular packages without need for bundled or external libs 
- popular packages with libs in lib
- uncommon packages without libs
- uncommon packages with libs
- accelerated packages
so if you'd loop over the lists of packages in this order, and then collect the list of  .cpp files in each of them. chances are high that you hit a duplicate file quickly and don't have to glob through all files and folders. if packages are installed, it is probable that packages like MOLECULE, MANYBODY, KSPACE, or RIGID are among them. so if those are checked early, you can reduce the cost of the globbing

BTW: the same issue affects collecting the list of installed packages with the makefile build. using the hidden file may accelerate that procedure as well. but let us first try it out as above (or as in my pending PR) and wait until people complain thinking that a machine is hung.

Reply:
Meh, that is too much heuristics for me! Plus, this doesn't really help much for the most common case that there is no conflict!

Reply:
good point. my bad.

here is a final suggestion. it won't speed up the first check, but you can create a time stamp file at the end of the check and then you only need to check if there is a .cpp or .h file in src/ that is newer than the time stamp file. I just added a similar optimization in PR #899 to only do the costly operation of compiling the list of installed packages when there is a newer file in src/.

Reply:
I think you are right. The time stamp file is just me thinking "make", where there is no checksum option and comparing time stamps is the native way to trigger actions.